devnet-sdk/testing/testlib/validators/validators_test.go

```diff
@@ -70,6 +86,189 @@ func TestValidators(t *testing.T) {
+					config: &mockChainConfig{
+						activeForks: map[rollup.ForkName]uint64{
+							rollup.Isthmus: 150, // Activates after current timestamp
+						},
+						currentTimestamp: 100,
+					},
+				},
+			},
+		}
+
+		// Get the validator for requiring Isthmus fork to not be active
+		chainConfigGetter, validator := AcquireL2WithoutFork(0, rollup.Isthmus)
+		systestT := systest.NewT(t)
+
+		// Apply the validator
+		ctx, err := validator(systestT, systestSystem)
+		require.NoError(t, err, "Validator should pass when fork is not active")
+
+		// Verify the chain config getter works
+		chainConfig := chainConfigGetter(ctx)
+		require.NotNil(t, chainConfig)
+		require.False(t, chainConfig.IsForkActivated(rollup.Isthmus, 100))
+	})
+
+	t.Run("test AcquireRequiresNotL2Fork - fork active", func(t *testing.T) {
+		// Create a system with the Isthmus fork active
+		systestSystem := &mockSystem{
+			l2s: []system.Chain{
+				&mockChain{
+					config: &mockChainConfig{
+						activeForks: map[rollup.ForkName]uint64{
+							rollup.Isthmus: 50,
+						},
+						currentTimestamp: 100, // After fork activation
+					},
+				},
+			},
+		}
+
+		// Get the validator for requiring Isthmus fork to not be active
+		_, validator := AcquireL2WithoutFork(0, rollup.Isthmus)
+		systestT := systest.NewT(t)
+
+		// Apply the validator - should fail since fork is active
+		_, err := validator(systestT, systestSystem)
+		require.Error(t, err, "Validator should fail when fork is active")
+		require.Contains(t, err.Error(), "has fork", "Error message should indicate fork is active")
+	})
+
+	t.Run("test AcquireRequiresExactL2Fork - exact fork", func(t *testing.T) {
+		// Create a system with exactly the Isthmus fork active (and no later forks)
+		systestSystem := &mockSystem{
+			l2s: []system.Chain{
+				&mockChain{
+					config: &mockChainConfig{
+						activeForks: map[rollup.ForkName]uint64{
+							rollup.Isthmus: 50,   // Active
+							rollup.Canyon:  1000, // Not yet active
+						},
+						currentTimestamp: 100, // After Isthmus, before Canyon
+					},
+				},
+			},
+		}
+
+		// Get the validator for requiring exactly the Isthmus fork
+		chainConfigGetter, validator := AcquireL2WithExactFork(0, rollup.Isthmus)
+		systestT := systest.NewT(t)
+
+		// Apply the validator
+		ctx, err := validator(systestT, systestSystem)
+		require.NoError(t, err, "Validator should pass when exactly at the specified fork")
+
+		// Verify the chain config getter works
+		chainConfig := chainConfigGetter(ctx)
+		require.NotNil(t, chainConfig)
+		require.Equal(t, rollup.Isthmus, chainConfig.CurrentFork(100))
+	})
+
+	t.Run("test AcquireRequiresExactL2Fork - later fork active", func(t *testing.T) {
+		// Create a system where a later fork (Canyon) is already active
+		systestSystem := &mockSystem{
+			l2s: []system.Chain{
+				&mockChain{
+					config: &mockChainConfig{
+						activeForks: map[rollup.ForkName]uint64{
+							rollup.Isthmus: 50,
+							rollup.Canyon:  75,
```

Comment:
ditto

Reply:
Removed as part of removing `AcquireL2WithExactFork()`